openslides_backend/action/mediafile/move.py

```diff
@@ -0,0 +1,87 @@
+from typing import List, Optional
+
+from ...models.models import Mediafile
+from ...shared.exceptions import ActionException
+from ...shared.filters import FilterOperator
+from ...shared.patterns import FullQualifiedId
+from ...shared.schema import id_list_schema
+from ..base import ActionPayload
+from ..default_schema import DefaultSchema
+from ..generics import UpdateAction
+from ..register import register_action
+
+
+@register_action("mediafile.move")
+class MediafileMoveAction(UpdateAction):
+    """
+    Action to move mediafiles.
+    """
+
+    model = Mediafile()
+    schema = DefaultSchema(Mediafile()).get_default_schema(
+        title="Mediafile new parent schema",
+        description="An object containing an array of mediafile ids and the new parent id the items should be moved to.",
+        required_properties=["parent_id", "meeting_id"],
+        additional_required_fields={
+            "ids": {
+                "description": "An array of agenda item ids where the items should be assigned to the new parent id.",
+                **id_list_schema,
+            }
+        },
+        single_item=True,
+    )
+
+    def check_is_directory(self, id_: int) -> None:
+        item = self.datastore.get(
+            FullQualifiedId(self.model.collection, id_), ["is_directory"]
+        )
+        if not item.get("is_directory"):
+            raise ActionException("New parent is not a directory.")
+
+    def get_updated_instances(self, payload: ActionPayload) -> ActionPayload:
+        # Payload is an iterable with exactly one item
+        instance = next(iter(payload))
+        yield from self.prepare_assign_data(
+            parent_id=instance["parent_id"],
+            ids=instance["ids"],
+            meeting_id=instance["meeting_id"],
+        )
+
+    def prepare_assign_data(
+        self, parent_id: Optional[int], ids: List[int], meeting_id: int
+    ) -> ActionPayload:
+        filter = FilterOperator("meeting_id", "=", meeting_id)
+        db_instances = self.datastore.filter(
+            collection=self.model.collection,
+            filter=filter,
+            mapped_fields=["id"],
+            lock_result=True,
+        )
+
+        if parent_id is None:
+            for id_ in ids:
+                if id_ not in db_instances:
+                    raise ActionException(f"Id {id_} not in db_instances.")
+                yield {"id": id_, "parent_id": None}
```

Comment:
This is the same as the `else` case, you can just move this behind the if-else

Reply:
Moved it.